File: android-native/app/src/main/java/com/scenicroutes/app/data/model/Subscription.kt
package com.scenicroutes.app.data.model

data class Subscription(
    val id: Long? = null,
    val user_id: Long? = null,
    val plan: String? = null, // "free", "premium", "pro"
    val status: String? = null, // "active", "cancelled", "expired"
    val payment_method: String? = null,
    val external_subscription_id: String? = null, // Google Play purchase token
    val product_id: String? = null, // Google Play product ID (e.g., "scenic_routes_premium_monthly")
    val platform: String? = null, // "google_play" for Android
    val billing_cycle: String? = null, // "monthly", "yearly"
    val amount: Double? = null,
    val currency: String? = null,
    val starts_at: String? = null,
    val ends_at: String? = null,
    val trial_ends_at: String? = null,
    val cancelled_at: String? = null,
    val cancellation_reason: String? = null,
    val cancel_at_period_end: Boolean? = null,
    val metadata: Map<String, Any>? = null,
)

File: app/Models/Subscription.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Subscription extends Model
{
    protected $fillable = [
        'user_id',
        'plan', // 'free', 'premium', 'pro'
        'stripe_subscription_id',
        'stripe_customer_id',
        'stripe_price_id',
        'payment_method',
        'billing_cycle', // 'monthly', 'yearly'
        'amount',
        'currency',
        'starts_at',
        'ends_at',
        'trial_ends_at',
        'cancelled_at',
        'cancellation_reason',
        'cancel_at_period_end',
        'status', // 'active', 'cancelled', 'expired'
        'metadata',
    ];

    protected $casts = [
        'starts_at' => 'datetime',
        'ends_at' => 'datetime',
        'trial_ends_at' => 'datetime',
        'cancelled_at' => 'datetime',
        'cancel_at_period_end' => 'boolean',
        'amount' => 'decimal:2',
        'metadata' => 'array',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
